src/osd/PrimaryLogPG.cc

```diff
@@ -5655,7 +5655,10 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 		 op.extent.length, op.extent.truncate_size,
 		 op.extent.truncate_seq);
       if (op_finisher == nullptr) {
-	result = do_sparse_read(ctx, osd_op);
+	if (osd->store->get_type() == "bluestore")
+	  result = do_read(ctx, osd_op);
```

Comment:
do_sparse_read will return the fiemap too[1], so this will break the backward compatibility.

-----------------------------------------
[1] https://github.com/ceph/ceph/blob/master/src/osd/PrimaryLogPG.cc#L5477

Reply:
Are you mean fill the hole is done by client rather than objectstore?

Reply:
No, I mean this change will break the API compatibility. do_sparse_read will return both fiemap and the contents, while do_read does not.

Reply:
Thanks @xiexingguo.